src/app/admin/modules/data-management/components/data-graph-options/data-graph-options.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, OnInit, Output, EventEmitter } from '@angular/core';
+import { FormBuilder, Validators, ValidatorFn, FormGroup, ValidationErrors} from '@angular/forms';
+import { OptionType } from 'src/app/shared/interfaces/option-type';
+import { Department } from 'src/app/shared/interfaces/department';
+import { CanvasOptionsService } from 'src/app/shared/services/data/canvas-options.service';
+import { DepartmentService } from 'src/app/shared/services/department.service';
+import { ListRequest } from 'src/app/shared/interfaces/list-request';
+import { DataGraphOption } from 'src/app/shared/interfaces/data-graph-option';
+
+export const timeValidator: ValidatorFn = (control: FormGroup): ValidationErrors | null => {
+    const selectedStartYear = control.get('selectedStartYear');
+    const selectedEndYear = control.get('selectedEndYear');
+
+    return selectedStartYear.value && selectedEndYear.value && selectedStartYear.value > selectedEndYear.value ?
+           { timeValidator: true } : null;
+};
+const WHOLE_SCHOOL = 0;
+
+@Component({
+  selector: 'app-data-graph-options',
+  templateUrl: './data-graph-options.component.html',
+  styleUrls: ['./data-graph-options.component.css']
+})
+export class DataGraphOptionsComponent implements OnInit {
+
+  @Output() getOptions = new EventEmitter();
+  selectedGraph: FormGroup;
+  showDepartmentSelector = true;
+  yearList: number[] = [];
+
+  departmentsList: Department[] = [{id: 0, name: '全校'} as Department];
+  statisticsType: OptionType[];
+
+  SelectedParamChangingCheck() {
+    this.selectedGraph.get('selectedStatisticsType').valueChanges.subscribe(val => {
+      if (val === null) {
+        return;
+      }
+      this.showDepartmentSelector = this.statisticsType[val]
+        .key === 'TRAINING_HOURS_WORKLOAD_STATISTICS' ? false : true;
+      this.selectedGraph.patchValue({
+        selectedGroupType: null,
+        selectedDepartment: WHOLE_SCHOOL
+      });
+    });
+    this.selectedGraph.get('selectedGroupType').valueChanges.subscribe(val => {
+      if (val === null) {
+        return;
+      }
+      const first = this.selectedGraph.get('selectedStatisticsType').value;
+      const second = this.selectedGraph.get('selectedGroupType').value;
+      if (this.statisticsType[first].key === 'TRAINING_HOURS_WORKLOAD_STATISTICS' ||
```

Comment:
Consider defining a enum mapping the backend definition for such as `TRAINING_HOURS_WORKLOAD_STATISTICS`, so you don't have to change many occurrences when the key content is changed.